pkg/operator/controller/status/controller_test.go

```diff
@@ -664,3 +664,140 @@ func Test_computeOperatorEvaluationConditionsDetectedCondition(t *testing.T) {
+				Status: configv1.ConditionTrue,
+				Reason: "IngressDoesNotExist",
+			},
+		},
+		{
+			description: "no default ingresscontroller",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("test1", false),
+					icWithStatus("test2", false),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorDegraded,
+				Status: configv1.ConditionTrue,
+				Reason: "IngressDoesNotExist",
+			},
+		},
+		{
+			description: "default ingresscontroller degraded",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", true),
+					icWithStatus("test", false),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorDegraded,
+				Status: configv1.ConditionTrue,
+				Reason: "IngressDegraded",
+			},
+		},
+		{
+			description: "default ingresscontroller not degraded",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", false),
+					icWithStatus("test2", true),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorDegraded,
+				Status: configv1.ConditionFalse,
+				Reason: "IngressNotDegraded",
+			},
+		},
+		{
+			description: "default ingresscontroller status unknown",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					ic("default"),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorDegraded,
+				Status: configv1.ConditionUnknown,
+				Reason: "IngressDegradedStatusUnknown",
+			},
+		},
+		{
+			description: "default ingresscontroller not degraded but unmanaged gateway api crds exist",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", false),
+				},
+				unmanagedGatewayAPICRDNames: "listenersets.gateway.networking.x-k8s.io",
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorDegraded,
+				Status: configv1.ConditionTrue,
+				Reason: "GatewayAPICRDsDegraded",
+			},
+		},
+		{
+			description: "default ingresscontroller degraded and unmanaged gateway api crds exist",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", true),
+				},
+				unmanagedGatewayAPICRDNames: "listenersets.gateway.networking.x-k8s.io",
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorDegraded,
+				Status: configv1.ConditionTrue,
+				Reason: "IngressDegradedAndGatewayAPICRDsDegraded",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			actual := computeOperatorDegradedCondition(tc.state)
+			cmpOpts := []cmp.Option{
+				cmpopts.IgnoreFields(configv1.ClusterOperatorStatusCondition{}, "Message"),
```

Comment:
It would be worth testing at least one case to make sure the Message is concatenated properly.

Reply:
Done.